web/src/test/java/com/replica/RouterServerTest.java

```diff
@@ -306,6 +324,167 @@ public void testIntraFeedPublicTransitQuery() {
+        RouterOuterClass.PtPath path = response.getPaths(0);
+        List<RouterOuterClass.PtLeg> streetLegs = path.getLegsList().stream()
+                .filter(l -> !l.hasTransitMetadata()).collect(Collectors.toList());
+        List<RouterOuterClass.PtLeg> ptLegs = path.getLegsList().stream()
+                .filter(RouterOuterClass.PtLeg::hasTransitMetadata).collect(Collectors.toList());
+        assertEquals(1, ptLegs.size());
+        assertEquals(2, streetLegs.size()); // access and egress
+        assertTrue(path.getDistanceMeters() > 0);
+        assertTrue(path.getDurationMillis() > 0);
+
+        // Check that street legs contain proper info
+        List<String> observedTravelSegmentTypes = Lists.newArrayList();
+        List<String> expectedTravelSegmentTypes = Lists.newArrayList("ACCESS", "EGRESS");
+        Map<String, Integer> observedModeCounts = Maps.newHashMap();
+        observedModeCounts.put("car", 0);
+        observedModeCounts.put("foot", 0);
+        Map<String, Integer> expectedModeCounts = Maps.newHashMap();
+        expectedModeCounts.put("car", 1);
+        expectedModeCounts.put("foot", 1);
+        List<String> observedStableEdgeIds = Lists.newArrayList();
+        int observedStableEdgeIdCount = 0;
+        double observedDistanceMeters = 0;
+        for (RouterOuterClass.PtLeg streetLeg : streetLegs) {
+            assertTrue(streetLeg.getStableEdgeIdsCount() > 0);
+            observedStableEdgeIdCount += streetLeg.getStableEdgeIdsCount();
+            observedStableEdgeIds.addAll(streetLeg.getStableEdgeIdsList());
+            assertTrue(streetLeg.getArrivalTime().getSeconds() > streetLeg.getDepartureTime().getSeconds());
+            assertTrue(streetLeg.getDistanceMeters() > 0);
+            assertFalse(streetLeg.getTravelSegmentType().isEmpty());
+            observedTravelSegmentTypes.add(streetLeg.getTravelSegmentType());
+            observedModeCounts.put(streetLeg.getMode(), observedModeCounts.get(streetLeg.getMode()) + 1);
+            observedDistanceMeters += streetLeg.getDistanceMeters();
+        }
+        assertEquals(expectedTravelSegmentTypes, observedTravelSegmentTypes);
+        assertEquals(expectedModeCounts, observedModeCounts);
+
+        // Check that PT legs contains proper info
+        for (RouterOuterClass.PtLeg ptLeg : ptLegs) {
+            assertTrue(ptLeg.getArrivalTime().getSeconds() > ptLeg.getDepartureTime().getSeconds());
+            assertTrue(ptLeg.getStableEdgeIdsCount() > 0); // check that the GTFS link mapper worked
+            observedStableEdgeIdCount += ptLeg.getStableEdgeIdsCount();
+            observedStableEdgeIds.addAll(ptLeg.getStableEdgeIdsList());
+            assertTrue(ptLeg.getDistanceMeters() > 0);
+            observedDistanceMeters += ptLeg.getDistanceMeters();
+
+            RouterOuterClass.TransitMetadata ptMetadata = ptLeg.getTransitMetadata();
+            assertFalse(ptMetadata.getTripId().isEmpty());
```

Comment:
nit: worth pulling some of these checks into a method? saw they're also present in `testInterFeedPublicTransitQuery`. same with the `observedStableEdgeIdCount` and `observedDistanceMeters` logic

Reply:
Great call! Made one common method for doing all the checking and it cleaned this up loads 🎉